sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_common/_base_regex_eval.py

```diff
@@ -0,0 +1,183 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+from typing import Any, Dict, List, Optional
+from typing_extensions import override
+from abc import abstractmethod
+import re
+import numpy as np
+
+from azure.ai.evaluation._evaluators._common import EvaluatorBase
+
+
+class RegexEvaluatorBase(EvaluatorBase):
+    """Base class for all evaluators that are regex-based and use pattern matching to extract answers.
+
+    This class provides a framework for evaluators that need to extract structured answers from text
+    using regular expressions. It handles the common pattern of:
+    1. Using regex patterns to extract answers from predictions
+    2. Comparing extracted answers against expected answers
+    3. Computing accuracy and instruction-following metrics
+
+    Child classes must implement the abstract methods:
+        - extract_expected_answer: Extract the expected answer from the label
+        - extract_regex: Extract a match object from the prediction using regex patterns
+        - compute_match: Compare actual and extracted answers to determine correctness
+
+    :param regex_patterns: A list of regex patterns to use for extracting answers from predictions.
+        Each pattern should have a single capture group to extract the answer. If None, the child class
+        must implement get_regex_patterns method.
+    :type regex_patterns: Optional[List[str]]
+    """
+
+    def __init__(self, *, regex_patterns: Optional[list[str]] = None, threshold=0.5) -> None:
+        super().__init__(threshold=threshold, _higher_is_better=True)
+        self.regex_patterns = regex_patterns
+        self.is_missing_regex_patterns = regex_patterns is None
+        self.follow_instructions = []
+        self.scores = []
+        self.chain_of_thought_lengths = []
+
+    @override
+    async def _do_eval(self, eval_input: Dict) -> Dict[str, Any]:
+        """Produce a score evaluation result.
+
+        :param eval_input: The input to the evaluation function.
+        :type eval_input: Dict
+        :return: The evaluation result.
+        :rtype: Dict
+        """
+        response = eval_input["response"]
+        ground_truth = eval_input["ground_truth"]
+
+        result = self.update(
+                prediction=response,
+                label=ground_truth,
+                json_data={}
+            )
+
+        return result
+
+    def update(self, prediction: str, label: str, json_data: dict) -> Dict[str, Any]:
```

Comment:
Please start name of method with underscore if it is not be exposed to customers